src/react-hooks-nesting-walker/error-messages.ts

```diff
@@ -17,4 +17,6 @@ export const ERROR_MESSAGES = {
     'A hook cannot be used inside of another function',
   invalidFunctionExpression: 'A hook cannot be used inside of another function',
   hookAfterEarlyReturn: 'A hook should not appear after a return statement',
+  anonymousFunctionIllegalCallback:
+    'Hook is in an anonymous function that is passed to an illegal callback',
```

Comment:
I wonder if it woulld be more helpful if we provided more information on this violation, explaining what an illegal callback is. This would match [the behavior of the ESLint rule](https://github.com/facebook/react/blob/51947a14bb24bd151f76f6fc0acdbbc404de13f7/packages/eslint-plugin-react-hooks/src/RulesOfHooks.js#L482-L485).

I know the existing messages are pretty terse, but your point made in #29 about providing more helpful error messages made me think if having that extra bit of info would be helpful to users of this TSLint plugin

Reply:
I agree. Though I'm not familiar on how to properly format (new line, expected format, etc) and test that string as it got a bit wide.

Please let me know what do you think of it.